gobblin-core/src/main/java/org/apache/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -342,9 +346,16 @@ protected void publishData(WorkUnitState state, int branchId, boolean publishSin
 
       movePath(parallelRunner, state, writerOutputDir, publisherOutputDir, branchId);
       writerOutputPathsMoved.add(writerOutputDir);
+      addPublisherLineageInfo(state, branchId, publisherOutputDir.toString());
     }
   }
 
+  protected void addPublisherLineageInfo(WorkUnitState state, int branchId, String output) {
+    LineageInfo info = LineageInfo.load(state, LineageInfo.Level.TASK_LEVEL);
```

Comment:
Is there a need to deserialize the existing `LineageInfo`? Can't you just add the new stuff without deserializing the old stuff?